ui-core/src/components/inputs/ComboBox/ComboBox.tsx

```diff
@@ -70,21 +71,28 @@ const InputWithSuggestions = <T,>({
           {
             icon: <ChevronDown size={small ? 'sm' : 'lg'} />,
             action: focusInput,
-            className: 'chevron-icon',
+            className: cx('chevron-icon', {
+              disabled: inputProps.disabled,
+            }),
           },
         ]
       : []), // Conditionally include the chevron icon only when suggestions are not empty
   ];
 
   useEffect(() => {
     if (value) {
-      setInputValue(getSuggestionLabel(value));
+      setInputValue(value);
     }
-  }, [value, getSuggestionLabel]);
+  }, [value]);
+
+  useEffect(() => {
+    setFilteredSuggestions(sortedSuggestions);
+  }, [suggestions, sortedSuggestions]);
```

Comment:
`sortedSuggestions` will already be re-triggered when `suggestions` changes (in the useMemo).

```suggestion
  }, [sortedSuggestions]);
```

Reply:
suggested change added :)